docs/rest-api.md

```diff
@@ -322,6 +323,70 @@ whitelist
 
 ```
 
+### Message WebSocket
+
+The API Server makes available a websocket endpoint for subscribing to RPC messages
+from the FreqTrade Bot. This can be used to consume real-time data from your bot, such as entry/exit fill messages, whitelist changes, populated indicators for pairs, and more.
+
+Assuming your rest API is set to `127.0.0.1` on port `8080`, the endpoint is available at `http://localhost:8080/api/v1/message/ws`.
+
+To access the websocket endpoint, the `ws_token` is required as a query parameter in the endpoint URL. This is set in your rest API config section. To generate a safe `ws_token` you can run the following code:
+
+``` python
+>>> import secrets
+>>> secrets.token_urlsafe(16)
+'zs9XYCbTPKvux46UJckflw'
+```
+
+You could then connect to the endpoint at `http://localhost:8080/api/v1/message/ws?token=zs9XYCbTPKvux46UJckflw`.
+
+!!! warning "Warning"
+
+    Please do not use the above example token. To make sure you are secure, generate a completely new token.
+
+
+#### Using the WebSocket
+
+Once connected to the WebSocket, the bot will broadcast RPC messages to anyone who is subscribed to them. To subscribe to a list of messages, you must send a JSON request through the WebSocket like the one below. The `data` key must be a list of message type strings.
+
+``` json
+{
+  "type": "subscribe",
+  "data": ["whitelist", "analyzed_df"] // A list of string message types
+}
+```
+
+#### Message types
+
+| Message Type | Description |
+|--------------|-------------|
+| whitelist | The list of pairs in the bot's whitelist.
+| analyzed_df | The dataframe and last_analyzed datetime for a pair.
+| entry | Trade has signaled an entry
+| entry_fill | Trade enter has filled
+| entry_cancel | Trade enter has been canceled
+| exit | Trade has signaled an exit
+| exit_fill | Trade exit has filled
+| exit_cancel | Trade exit has been canceled
+| protection_trigger | A protection has triggered for a pair
+| protection_trigger_global | A protection has triggered for a pair
+| status | A bot's status change
+| startup | Startup messages
+| warning | Any warnings
```

Comment:
i'm ... tempted to not have these here - but instead direct people to the code.

As we've seen in other places it'll become outdated pretty quickly (this list is in a similar form in webhooks - and most likely also in telegram.md).
webhook subscribtion is however not something that will be an "everyday user task" - so requiring a developer implementing this to look at the enum should be fine